src/vllm_router/README.md

```diff
@@ -116,21 +118,46 @@ Currently, the dynamic config supports the following fields:
 
 **Optional fields:**
 
+- `callbacks`: The path to the callback instance extending CustomCallbackHandler.
 - (When using `static` service discovery) `static_backends`: The URLs of static serving engines, separated by commas (e.g., `http://localhost:9001,http://localhost:9002,http://localhost:9003`).
 - (When using `static` service discovery) `static_models`: The models running in the static serving engines, separated by commas (e.g., `model1,model2`).
+- (When using `static` service discovery and if you enable the `--static-backend-health-checks` flag) `static_model_types`: The model types running in the static serving engines, separated by commas (e.g., `chat,chat`).
 - (When using `k8s` service discovery) `k8s_port`: The port of vLLM processes when using K8s service discovery. Default is `8000`.
 - (When using `k8s` service discovery) `k8s_namespace`: The namespace of vLLM pods when using K8s service discovery. Default is `default`.
 - (When using `k8s` service discovery) `k8s_label_selector`: The label selector to filter vLLM pods when using K8s service discovery.
 - `session_key`: The key (in the header) to identify a session when using session-based routing.
 
-Here is an example dynamic config file:
+Here is an example of a dynamic YAML config file:
+
+```yaml
+service_discovery: static
+routing_logic: roundrobin
+callbacks: module.custom.callback_handler
+static_models:
+    facebook/opt-125m:
+        static_backends:
+            - http://localhost:9001
+        static_model_type: completion
+    meta-llama/Llama-3.1-8B-Instruct:
+        static_backends:
+            - http://localhost:9002
+        static_model_type: chat
+    facebook/opt-125m:
```

Comment:
We can have extra nested keys/values for each of the `static_models` because we pop the `static_models` object before processing it, see other comment.

So this YAML config would also be accepted:

```yaml
service_discovery: static
routing_logic: roundrobin
callbacks: module.custom.callback_handler
static_models:
    facebook/opt-125m:
        static_backends:
            - http://localhost:9003
        static_model_type: completion
        license: Apache-2.0
        note: "This is an extra note"
```